### ISSUE TITLE
Issue with operator 1.2.32 when replicaCount > 1

### ISSUE THREAD

**Issue by jgoodall** [arangodb/kube-arangodb#1388]
kube-arangodb 1.2.31 operator works fine for us, but 1.2.32 fails to start when the operator is configured with `operator.replicaCount=2` (using the helm chart). The operators instances report this error: `FTL Failed to create resource lock error="endpoints lock is removed, migrate to endpointsleases"`. Setting `replicaCount=1` or leaving that setting out completely "fixes" the problem (since there is no leader). Related note - the helm documentation says the default is 2, but the actual default is 1). 

This issue seems to stem from the change made to `pkg/operator/operator_leader.go` in commit c89b45f.

Version info:

```
$ kubectl version --output yaml
clientVersion:
  buildDate: "2023-08-15T10:15:49Z"
  compiler: gc
  gitCommit: 855e7c48de7388eb330da0f8d9d2394ee818fb8d
  gitTreeState: clean
  gitVersion: v1.28.0
  goVersion: go1.20.7
  major: "1"
  minor: "28"
  platform: darwin/arm64
kustomizeVersion: v5.0.4-0.20230601165947-6ce0bf390ce3
serverVersion:
  buildDate: "2023-07-28T09:46:05Z"
  compiler: gc
  gitCommit: 36645e7311e9bdbbf2adb79ecd8bd68556bc86f6
  gitTreeState: clean
  gitVersion: v1.27.4+k3s1
  goVersion: go1.20.6
  major: "1"
  minor: "27"
  platform: linux/arm64
$ helm version
version.BuildInfo{Version:"v3.12.3", GitCommit:"3a31588ad33fe3b89af5a2a54ee1d25bfe6eaa5e", GitTreeState:"clean", GoVersion:"go1.20.7"}
```

**Comment by ajanikow** [comment]
Hello!

We are aware of this issue, the fix is in progress.

FYI @jwierzbo 

Best,
Adam.

**Comment by jwierzbo** [comment]
It is already fixed in master https://github.com/arangodb/kube-arangodb/commit/2e2333b412cd738eecc11e97dcc6e0edc5f56ab3

**Comment by jgoodall** [comment]
I didnt see this in the list of changes on the release page - did this get addressed in 1.2.33?

**Comment by jwierzbo** [comment]
Yes: https://github.com/arangodb/kube-arangodb/commit/2e2333b412cd738eecc11e97dcc6e0edc5f56ab3
